### ISSUE TITLE
SIGSEGV: Illegal storage access in asynchttpserver.

### ISSUE THREAD

**Issue by korakon** [nim-lang/Nim#13937]
Run the example provided below and make a simple request to the server. it will crash the server.

```bash
> nim c -r example.nim
> curl localhost:8080
```

### Example
```nim
import asynchttpserver, asyncdispatch

var server = newAsyncHttpServer()
proc cb(req: Request) {.async.} =
  await req.respond(Http200, "Hello World")

echo "Listening on port 8080"
waitFor server.serve(Port(8080), cb)
```

### Current Output
```
Traceback (most recent call last)
/tmp/nim-bug/main.nim(7) main
/usr/lib/nim/pure/asyncdispatch.nim(1886) waitFor
/usr/lib/nim/pure/asyncdispatch.nim(1576) poll
/usr/lib/nim/pure/asyncdispatch.nim(1340) runOnce
/usr/lib/nim/pure/asyncdispatch.nim(210) processPendingCallbacks
/usr/lib/nim/pure/asyncmacro.nim(37) processRequestNimAsyncContinue
/usr/lib/nim/pure/asynchttpserver.nim(258) processRequestIter
/usr/lib/nim/pure/asyncmacro.nim(319) cb
/usr/lib/nim/pure/asyncmacro.nim(34) cbNimAsyncContinue
/tmp/nim-bug/main.nim(5) cbIter
/usr/lib/nim/pure/asynchttpserver.nim(103) respond
/usr/lib/nim/pure/httpcore.nim(180) hasKey
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
Error: execution of an external program failed: '/tmp/nim-bug/main '
```

### Expected Output
```
Hello world
```

### Additional Information

* It was working in version 1.0.6

```
$ nim -v
Nim Compiler Version 1.2.0 [Linux: amd64]
Compiled at 2020-04-04
Copyright (c) 2006-2020 by Andreas Rumpf

active boot switches: -d:release -d:nativeStackTrace
```

**Comment by kubo39** [comment]
@korakon Perhaps fixed at https://github.com/nim-lang/Nim/pull/13867

**Comment by soufiane** [comment]
> @korakon Perhaps fixed at #13867

Yes, I think that fixes it.

**Comment by korakon** [comment]
> @korakon Perhaps fixed at #13867

I think that fixes it, I wasn't aware of that pull request before creating this issue, thank you for pointing me to it.

**Comment by Araq** [comment]
Fix will be in 1.2.2.
